pkg/discovery/network/openshift4.go

```diff
@@ -90,13 +95,17 @@ func parseOS4Network(cr *unstructured.Unstructured) (*ClusterNetwork, error) {
 		result.ServiceCIDRs = append(result.ServiceCIDRs, serviceNetwork.(string))
 	}
 
-	result.NetworkPlugin, found, err = unstructured.NestedString(cr.Object, "spec", "networkType")
-
+	ocpNetworkType, found, err := unstructured.NestedString(cr.Object, "spec", "networkType")
 	if err != nil {
 		return nil, errors.Wrap(err, "error retrieving spec.networkType field")
 	} else if !found {
 		return nil, fmt.Errorf("field .spec.networkType expected, but not found in Network resource: %v", cr.Object)
 	}
-
+	// calico plugin named 'Calico' in networkType
+	if ocpNetworkType == NetworkTypeCalico {
```

Comment:
What if some users configure "calico" instead of "Calico"? Can we do case insensitive comparision here for Calico and configure the `result.NetworkPlugin = constants.NetworkPluginCalico`

Reply:
If some user will configure "calico" I think we should be OK (we'll fail to the else statement).

This extra code should map OCP NetworkType to the correct NetworkPlugin,
so maybe we should at least raise an error if network-discovery for OCP sets result.NetworkPlugin to value != one of the supported plugins, what do you think?

Reply:
> If some user will configure "calico" I think we should be OK (we'll fail to the else statement).

Oh yes, I missed it.

> This extra code should map OCP NetworkType to the correct NetworkPlugin, so maybe we should at least raise an error if network-discovery for OCP sets result.NetworkPlugin to value != one of the supported plugins, what do you think?

Such validations will be done using `subctl diagnose ...` tool